docs/annotating_code/type_syntax/utility_types.md

```diff
@@ -0,0 +1,40 @@
+# Utility types
```

Comment:
Where are `key-of` and `value-of` documented? Maybe they should be added here?

Reply:
I can add this as well in another PR.

Reply:
I added `key-of` and `value-of`, `SomeClass::CONST_*` also could need some documentation, but I think it's not a utility type if you look at typescripts definition (https://www.typescriptlang.org/docs/handbook/utility-types.html)